phoenix-core/src/it/java/org/apache/phoenix/monitoring/PhoenixMetricsIT.java

```diff
@@ -1298,5 +1301,225 @@ public Connection call() throws Exception {
         }
     }
 
+    @Test
+    public void testPhoenixClientQueueWaitTimeAndEndToEndTime() throws SQLException,
+            NoSuchFieldException,
+            IllegalAccessException {
+        int saltBucketNum = 8;
+        String tableName = generateUniqueName();
+        String getRows = "SELECT COL1, COL2, PK4, PK2, PK3 FROM " + tableName
+                + " WHERE PK1=? AND PK2=? AND PK3=? AND PK4 IN (?";
+        // Send at least 4 tasks in Phoenix client thread pool and we will assert also on this
+        // later while actually querying data.
+        int taskCount = 4;
+        for (int i = 1; i < taskCount; i++) {
+            getRows += ", ?";
+        }
+        getRows += ")";
+        final String upsertRows = "UPSERT INTO " + tableName + " VALUES(?, ?, ?, ?, ?, ?)";
+        String creatTableDdl = "CREATE TABLE IF NOT EXISTS " + tableName + " (\n" +
```

Comment:
Generally we put the table schema first and then the upsert statements and the select statements. This makes it easy to understand the code.